ui/components/app/snaps/snap-ui-markdown/snap-ui-markdown.js

```diff
@@ -16,22 +22,65 @@ const Paragraph = (props) => (
   />
 );
 
+const Link = ({ onClick, children, ...rest }) => (
+  <ButtonLink
+    {...rest}
+    onClick={onClick}
+    externalLink
+    size={ButtonLinkSize.Inherit}
+    endIconName={IconName.Export}
+    className="snap-ui-markdown__link"
+  >
+    {children}
+  </ButtonLink>
+);
+
 export const SnapUIMarkdown = ({ children, markdown }) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [redirectUrl, setRedirectUrl] = useState(undefined);
+
   if (markdown === false) {
     return <Paragraph>{children}</Paragraph>;
   }
 
+  const handleLinkClick = (url) => {
+    setRedirectUrl(url);
+    setIsModalOpen(true);
```

Comment:
Wondering if it makes sense to just use `reidrectUrl` to determine if the modal should be opened?

Reply:
addressing this 👍